packages/guides/src/Nodes/CodeNode.php

```diff
@@ -22,6 +22,8 @@ class CodeNode extends TextNode
     /** @var int|null The line number to start counting from and display, or null to hide line numbers */
     private int|null $startingLineNumber = null;
 
+    protected string|null $caption = null;
```

Comment:
Please make properties private by default, we can always open them when needed. Making a property private will help us to maintain this class.  As we are sure nobody is using it by extending the class as they do not have access to it.

When you define it as protected the property can not be changed any more without introducing a backwards compatibility break. Which makes it harder for people to upgrade, at least that's what they think. And of course we want people to use the latest version of this library, this might reduce the time we need to spend on invalid bug reports.